releases/v0.23.md

```diff
@@ -0,0 +1,192 @@
+# v0.23 🚀
+
+This document contains all release notes pertaining to the `v0.23.x` releases of
+Vela.
+
+## For Vela Administrators
+
+- When upgrading from `v0.23`, please take note of the [migration information](/migrations/v0.23/README.md)
+
+## v0.23.0
+
+### 📣 Highlights
+
+- Allowed Events
+- Deployment records
+- Improved Local Pipeline Execution Experience
+- Improved Pipeline Visualization
+- Support Build Approval
+- Bug Fixes and other Enhancements
+
```

Comment:
```suggestion

- Allowed Events
- Native Deployment Records
- Introducing Build Approval
- Bug Fixes and other Enhancements

```

looks like some commits from previous release made it into the list below. let's rerun after final version is cut